op-supervisor/supervisor/backend/backend.go

```diff
@@ -151,6 +154,10 @@ func (su *SupervisorBackend) initResources(ctx context.Context, cfg *config.Conf
 		chainProcessor := processors.NewChainProcessor(su.logger, chainID, logProcessor, su.chainDBs, su.onIndexedLocalUnsafeData)
 		su.chainProcessors.Set(chainID, chainProcessor)
 	}
+	// initialize sync sources
+	for _, chainID := range chains {
+		su.syncSources.Set(chainID, nil)
+	}
```

Comment:
Don't we need to set these sync sources to something non-nil? Or are they always just attached after startup?

Reply:
The sync sources are attached during setup usually.

However tests don't have sync sources available during setup. Instead, sync sources are added dynamically. The nil syncSource entries here are to prime the mapping with a chainID. So that we can prevent a sync sources for unknown chains to be added later (i.e. check for existence of a sync source, nil or otherwise, at the provided chainID, before overwriting its sync source).